packages/core/src/controllers/relayer.ts

```diff
@@ -135,20 +148,50 @@ export class Relayer extends IRelayer {
 
   public async transportClose() {
     this.transportExplicitlyClosed = true;
-    await this.provider.disconnect();
+    if (this.connected) await this.provider.disconnect();
+    this.events.emit(RELAYER_EVENTS.transport_closed);
   }
 
   public async transportOpen(relayUrl?: string) {
     this.relayUrl = relayUrl || this.relayUrl;
     this.transportExplicitlyClosed = false;
-    await this.provider.connect();
-    // wait for the subscriber to finish resubscribing to its topics
-    await new Promise<void>((resolve) => {
-      this.subscriber.once(SUBSCRIBER_EVENTS.resubscribed, () => {
-        resolve();
-      });
-    });
+    try {
+      await Promise.race([
+        this.provider.connect(),
+        new Promise<void>((_res, reject) =>
+          // rejects pending promise if transport is closed before connection is established
+          // useful when .connect() gets stuck resolving
+          this.once(RELAYER_EVENTS.transport_closed, () => {
+            reject();
+          }),
+        ),
+      ]);
+
+      if (this.initialized) {
+        // wait for the subscriber to finish resubscribing to its topics
+        await new Promise<void>((resolve) => {
+          this.subscriber.once(SUBSCRIBER_EVENTS.resubscribed, () => {
+            resolve();
+          });
+        });
+      }
+    } catch (e: unknown | Error) {
+      const error = e as Error;
+      if (!/socket hang up/i.test(error.message)) {
+        throw new Error(error.message);
+      }
+      // eslint-disable-next-line no-console
+      console.error(error);
+      this.events.emit(RELAYER_EVENTS.transport_closed);
+    }
   }
+
+  public async restartTransport(relayUrl?: string) {
+    await this.transportClose();
+    await new Promise<void>((resolve) => setTimeout(resolve, RELAYER_RECONNECT_TIMEOUT));
```

Comment:
Can you explain this more please? Seems like it's effectively just doing a blocking `wait` for the duration of `RELAYER_RECONNECT_TIMEOUT`

Reply:
My reasoning was to give leeway for any replication delay in the relay and avoid the following scenario:
- initially connected to Node#1 
- disconnect transport
- reconnect transport immediately
- connected to node#2 before node#1 is done cleanup/replication etc

Reply:
Got it, thanks for context 👍 

I also just realised that `RELAYER_RECONNECT_TIMEOUT` is that standard one second we've always had for auto-reconnection attempts, so not as dramatic as it appeared at first.